src/pages/settings/Profile/Contacts/ContactMethodsPage.js

```diff
@@ -28,111 +33,136 @@ const propTypes = {
         partnerUserID: PropTypes.string,
 
         /** Date of when login was validated */
+        /** Date login was validated, used to show brickroad info status */
         validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+
+        /** Field-specific pending states for offline UI status */
+        pendingFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
     }),
 
+    /** Current user session */
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }).isRequired,
+
     ...withLocalizePropTypes,
-    ...withCurrentUserPersonalDetailsPropTypes,
 };
 
 const defaultProps = {
     loginList: {},
-    ...withCurrentUserPersonalDetailsDefaultProps,
 };
 
-class ContactMethodsPage extends Component {
-    constructor(props) {
-        super(props);
+const ContactMethodsPage = (props) => {
+    let hasPhoneNumberLogin = false;
+    let hasEmailLogin = false;
 
-        this.state = {
-            logins: this.getLogins(),
-        };
-
-        this.getLogins = this.getLogins.bind(this);
-    }
-
-    componentDidUpdate(prevProps) {
-        let stateToUpdate = {};
+    const loginMenuItems = _.map(props.loginList, (login, loginName) => {
+        const pendingAction = lodashGet(login, 'pendingFields.deletedLogin', null);
+        if (!login.partnerUserID && _.isEmpty(pendingAction)) {
+            return null;
+        }
 
-        // Recalculate logins if loginList has changed
-        if (_.keys(this.props.loginList).length !== _.keys(prevProps.loginList).length) {
-            stateToUpdate = {logins: this.getLogins()};
+        let description = '';
+        if (props.session.email === login.partnerUserID) {
+            description = props.translate('contacts.getInTouch');
+        } else if (!login.validatedDate) {
+            description = props.translate('contacts.pleaseVerify');
+        }
+        let indicator = null;
+        if (_.some(lodashGet(login, 'errorFields', {}), errorField => !_.isEmpty(errorField))) {
+            indicator = CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+        } else if (!login.validatedDate) {
+            indicator = CONST.BRICK_ROAD_INDICATOR_STATUS.INFO;
         }
 
-        if (_.isEmpty(stateToUpdate)) {
-            return;
+        // Temporary checks to determine if we need to show specific LoginField
+        // components. This check will be removed soon.
```

Comment:
There is a PR @cristipaval is working on that will allow adding multiple phone numbers. The PR is on hold while we get this one merged.

I think @cristipaval 's PR is [this one](https://github.com/Expensify/App/pull/15330), but he or @Beamanator probably know better about this

Reply:
Ok let's put that info in the comment then, that way it will be obvious if we somehow forget. Also the reader will have context about why it's going to be removed.

Reply:
Ok let's put that info in the comment then, that way it will be obvious if we somehow forget. Also the reader will have context about why it's going to be removed.

Reply:
Update comment adding a link to the follow up PR: https://github.com/Expensify/App/pull/15330